uniffi/src/testing.rs

```diff
@@ -117,25 +122,82 @@ pub fn ensure_compiled_cdylib(pkg_dir: &str) -> Result<String> {
 ///
 /// Internally, this function does a bit of caching and concurrency management to avoid rebuilding
 /// the component for multiple testcases.
-pub fn ensure_generated_bindings<P1, P2, L>(cdylib_file: P1, out_dir: P2, language: L) -> Result<()>
+pub fn ensure_generated_bindings<P1, P2>(
+    idl_file: P1,
+    out_dir: P2,
+    language_ext: &str,
+) -> Result<()>
 where
     P1: AsRef<Path>,
     P2: AsRef<Path>,
-    L: TryInto<bindings::TargetLanguage, Error = anyhow::Error>,
 {
-    let cdylib_file: PathBuf = cdylib_file.as_ref().to_path_buf();
+    let idl_file: PathBuf = idl_file.as_ref().to_path_buf();
     let out_dir: PathBuf = out_dir.as_ref().to_path_buf();
-    let language: bindings::TargetLanguage = language.try_into()?;
     // Have we already generated these bindings?
     let mut generated_bindings = GENERATED_BINDINGS.lock().unwrap();
     // XXX TODO: figure out correct type of `HashSet` so that I don't need to `clone()` here...
-    if generated_bindings.contains(&(cdylib_file.clone(), out_dir.clone(), language)) {
+    if generated_bindings.contains(&(idl_file.clone(), out_dir.clone(), language_ext.to_owned())) {
         return Ok(());
     }
     // Nope, looks like we'll have to compile it afresh.
-    let ci = bindings::get_component_interface_from_cdylib(&cdylib_file)?;
-    bindings::write_bindings(&ci, &out_dir, language)?;
-    bindings::compile_bindings(&ci, &out_dir, language)?;
-    generated_bindings.insert((cdylib_file, out_dir, language));
+
+    let status = Command::new("uniffi-bindgen")
+        .args(&[
+            "generate",
+            "--language",
+            ext_to_lang(language_ext)?,
+            "--out-dir",
+            out_dir.to_str().unwrap(),
+            idl_file.to_str().unwrap(),
+        ])
+        .status()?;
+    if !status.success() {
+        bail!("Error while generating bindings: {}",);
+    }
+
+    // XXX: technically, namespace != idl file name, but we don't
+    // want to take a dependency on `ComponentInterface` here.
+    let namespace = idl_file
+        .file_stem()
+        .ok_or_else(|| anyhow!("not a file"))?
+        .to_str()
+        .ok_or_else(|| anyhow!("not a string"))?;
```

Comment:
Two alternatives to relying on file-naming convention here:

* Do a half-hearted manual parse of the IDL file, just looking for a regex match on `namespace (\w+)`.
* Add another command to `uniffi-bindgen` to spit out component metadata from the IDL file.

Reply:
Alternately, maybe it would make sense to push some or all of the "compile and run foreign-language script against these bindings" logic out into a `uniffi-bindgen` subcommand as well? We already need `uniffi-bindgen` to exist to run the tests.